src/features/ci-cd/hooks/service-installs.ts

```diff
@@ -224,3 +224,138 @@ hooks.addPureHook('PATCH', 'resin', 'device', {
 		}
 	},
 });
+
+hooks.addPureHook('PATCH', 'resin', 'device', {
+	POSTRUN: async ({ api, request }) => {
+		const affectedIds = request.affectedIds!;
+		if (affectedIds.length === 0) {
+			return;
+		}
+
+		// Create supervisor serice installs when device checks in with it's initial version
+		if (request.values.supervisor_version) {
+			await createSupervisorServiceInstallsFromVersion(
+				api,
+				affectedIds,
+				request.values.supervisor_version,
+			);
+			return;
```

Comment:
This `return` would break things if setting `supervisor_version` and `should_be_managed_by__release` in the same request

Reply:
Never mind.. it works as you said.